native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -88,20 +92,37 @@ public class ArgumentHandler {
     private static final int T_BOOLEAN = 5;
     private static final int T_INPUT_OBJECT = 22;
     private static final int T_LIST = 23;
+    private static final ArrayList<String> idsList = new ArrayList<>();
+    private static final String ID_ANNOTATION = "ID";
 
     public ArgumentHandler(MethodType method, BObject context, BObject field, BObject responseGenerator,
                            boolean validation) {
         this.method = method;
         this.fileInfo = (BMap<BString, Object>) context.getNativeData(FILE_INFO_FIELD);
         this.context = context;
         this.field = field;
+        getIdTypeArguments(method.getAnnotations());
         this.argumentsMap = ValueCreator.createMapValue();
         this.responseGenerator = responseGenerator;
         this.validation = validation;
         BObject fieldNode = this.field.getObjectValue(INTERNAL_NODE);
         this.populateArgumentsMap(fieldNode);
     }
 
+    private void getIdTypeArguments(BMap<BString, Object> annotations) {
+        int i = 0;
+        for (Object annotation: annotations.values().toArray()) {
+            BMap annotationMap = (BMap) annotation;
+            for (Object annotationKey: annotationMap.getKeys()) {
+                  if (((BString) annotationKey).getValue().endsWith(ID_ANNOTATION)
+                        && !annotations.getKeys()[i].getValue().equals("$returns$")) {
+                      idsList.add(annotations.getKeys()[i].getValue().split("\\.")[1]);
+                      i++;
+                  }
+            }
```

Comment:
Can't we do this using runtime APIs instead of string manipulation?

Reply:
You can get the `AnnotatableType` from the annotation and then get the package name and the annotation name. (Need to check the type tag and cast to `AnnotatableType` first).